src/components/i18n/README.md

```diff
@@ -0,0 +1,296 @@
+    </span>
+</FormattedCompMessage>
+```
+
+The Param must have both a value and a description prop. The description describes for the translators what is substituted into the string.
+
+Various types of values are allowed in Param component values, including JSX and even functions! Example:
+
+```jsx
+// @NOTE: You can only use require instead of import in markdown.
+const Param = require('./Param').default;
+
+<FormattedCompMessage
+    id="unique.id"
+    description="description for translators to explain the context of this string"
+>
+    <span>
+        The type is <b><Param value="text" description="a textual parameter"/></b> which contains <span class="pretty-numbers"><Param value="2.334" description="number of items"/></span> items.
+        These items are <Param value={true} description="a boolean value"/>.
+        They contain <Param value={<span class="kotter">Any arbitrary <Link to="x">jsx</Link> can go here!</span>} description="arbitrary jsx"/>.
+        A function can return <Param description="the result of a call to a function" value={function() {
+            return "a value";
+        }}/>.
+    </span>
+</FormattedCompMessage>
+```
+
+Text is included as-is. Numbers are converted to strings using the regular javascript type coersion. It is highly recommended
+to use a number formatter instead, so that numbers are presented in a localized fashion. Booleans are converted to the string "true" or
+"false" in English. It is recommended to use the boolean value to choose between two localized strings instead of using these
+values in English. Jsx expressions get substituted in as-is. They are not translated unless you explicitly translate the text
+using intl.formatMessage(). If the value of a property is a function, that function will be called to retrieve
+a value to substitute in for the parameter.
+
+Parameters are represented in the string to translate as a self-closing tag of the form "p" + digits. This hides the contents
+of the parameter and allows React to rerender only the parts of the string that have changed when the value of the parameter
+component changes.
+
+For example, the string to translate for the above example would be:
+
+```
+"The type is <c0><p0/></c0> which contains <c1><p1/></c1> items. These items are <p2/>. They contain <p3/>. A function can return <p4/>."
+```
+
+Note that whitespace in the middle of the string is compressed because it doesn't matter to the translation or to the final
+HTML output.
+
+** Avoid This **
+
+Do not put brace expressions in the text to translate! They will get replaced before the FormattedCompMessage component
+is created, and therefore the source string will contain the value of the expression. When the component attempts to
+load the translation, that source string will not be found, and therefore the translation will also not be found.
+Strings with brace expressions in them will be rejected by the babel plugin when they are extracted.
+
```

Comment:
What is the error message returned to the user by babel plugin when they have curly braces instead of using `Param`? It should be descriptive enough so that they can fix it quickly.

Reply:
I'll make it more descriptive.

Reply:
Okay, the babel plugin is updated to mention how to fix it. (ie. use the Param tag instead.)